lms/static/scripts/frontend_apps/components/FilePickerApp.js

```diff
@@ -94,13 +98,37 @@ export default function FilePickerApp({ onSubmit }) {
   // Submit the form after a selection is made via one of the available
   // methods.
   useEffect(() => {
+    async function createAssignment() {
+      const data = { ...createAssignmentAPI.data };
```

Comment:
I get here:

```lms/static/scripts/frontend_apps/components/FilePickerApp.js:103:13 - error TS2322: Type '{ constructor?: Function | undefined; toString?: (() => string) | undefined; toLocaleString?: (() => string) | undefined; valueOf?: (() => Object) | undefined; hasOwnProperty?: ((v: PropertyKey) => boolean) | undefined; isPrototypeOf?: ((v: Object) => boolean) | undefined; propertyIsEnumerable?: ((v: PropertyKey) =>...' is not assignable to type 'Record<string, string | Content | null>'.
  Property 'constructor' is incompatible with index signature.
    Type 'Function' is not assignable to type 'string | Content | null'.
      Type 'Function' is not assignable to type 'string'.
```

I can figure out where the function come from, shouldn't createAssignmentAPI be of type `APICallInfo`

Reply:
Huh. It looks like this is a quirk to do with usage of the legacy `Object` type. The fix is:

```diff
diff --git a/lms/static/scripts/frontend_apps/config.js b/lms/static/scripts/frontend_apps/config.js
index e738df27..f5cc7d65 100644
--- a/lms/static/scripts/frontend_apps/config.js
+++ b/lms/static/scripts/frontend_apps/config.js
@@ -13,7 +13,7 @@ import { createContext } from 'preact';
  *   URL to display in a popup window if the call fails due to an authorization
  *   error. This is used when the API call requires an OAuth 2 authorization
  *   flow to be completed with eg. an external LMS's API
- * @prop {Object} [data]
+ * @prop {object} [data]
  */
```

Here `object` means "some value that is not a primitive (like a string or number), but we don't know what fields it might have. `Object` is a thing that exists for compatibility with existing JSDoc comments. @esanzgar has been busy changing it to `object` or `Record` recently.

`Object` is a thing that exists for compatibility with old JSDoc comments.